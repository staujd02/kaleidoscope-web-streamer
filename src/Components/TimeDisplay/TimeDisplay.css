.time-display.MuiButton-root{
    position: absolute;
    right: 15px;
    top: 15px;
    height: 70px;
    width: auto;
    padding-left: 20px;
    padding-right: 20px;
    background-color: rgba(0, 0, 0, .35);
    border-radius: 40px;
    cursor: pointer;
}

.time-display:hover{
    box-shadow: 0 0 7px 5px white;
    background-color: rgba(0, 0, 0, .65);
}

.time-box{
    color: rgba(200, 200, 200, .65);
    font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
    font-size: 24pt;
    text-align: center;
}