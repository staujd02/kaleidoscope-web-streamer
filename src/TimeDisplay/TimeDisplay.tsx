import React from 'react';
import './TimeDisplay.css';

class TimeDisplay extends React.Component<TimeDisplayProps>{

    render(){
        const { time } = this.props;
        const secondsRemaining = this.calcSeconds(time);
        const minutesRemaining = this.calcMinutes(time);
        return (
            <div className="time-display">
                <div className="time-box">{`${minutesRemaining}:${secondsRemaining}`}</div>
            </div>
        );
    }

    private calcMinutes(milSecondsRemaining: number) {
        return this.format(Math.floor(milSecondsRemaining / 60000));
    }

    private calcSeconds(milSecondsRemaining: number) {
        return this.format(Math.floor((milSecondsRemaining % 60000) / 1000));
    }

    private format(numberToFormat: number): string {
        var str = "" + numberToFormat;
        var pad = "00";
        return pad.substring(0, pad.length - str.length) + str;
    }
}

export default TimeDisplay;