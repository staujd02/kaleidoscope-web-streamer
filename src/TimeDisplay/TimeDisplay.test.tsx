import React from 'react';
import TimeDisplay from './TimeDisplay';
import renderer from 'react-test-renderer';

describe("The time display", () => {

    it('renders correctly', () => {
        let sixtyMinutesInMilli = 60 * 60 * 1000;
        let thirtySecondsInMilli = 3 * 1000;
        let combined = sixtyMinutesInMilli 
                + thirtySecondsInMilli;
        const tree = renderer
            .create(<TimeDisplay time={combined}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});