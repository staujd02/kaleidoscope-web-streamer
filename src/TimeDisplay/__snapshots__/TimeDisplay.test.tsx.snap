// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The time display renders correctly 1`] = `
<div
  className="time-display"
>
  <div
    className="time-box"
  >
    60:03
  </div>
</div>
`;
